astromodels/core/model_parser.py

```diff
@@ -346,15 +347,15 @@ def __init__(self, source_name, source_definition):
         # to make a synchrotron spectrum uses this to save and set up the particle distribution
         self._extra_setups = []
 
-        if source_type == POINT_SOURCE:
+        if source_type == SourceType.POINT_SOURCE.value:
 
             self._parsed_source = self._parse_point_source(source_definition)
 
-        elif source_type == EXTENDED_SOURCE:
+        elif source_type == SourceType.EXTENDED_SOURCE.value:
 
             self._parsed_source = self._parse_extended_source(source_definition)
 
-        elif source_type == PARTICLE_SOURCE:
+        elif source_type == SourceType.PARTICLE_SOURCE.value:
 
```

Comment:
I think this .value is not needed in lines 350, 354 and 358

Reply:
its need for string checking unfortunately

Reply:
Ah okay